flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -192,6 +196,18 @@ public String cancelWithSavepoint(JobID jobId, @Nullable String savepointDirecto
 			.thenApply(response -> response.location);
 	}
 
+	@Override
+	public CompletableFuture<Collection<JobDetails>> listJobs() throws Exception {
+		CurrentJobsOverviewHandlerHeaders headers = CurrentJobsOverviewHandlerHeaders.getInstance();
+		CompletableFuture<MultipleJobsDetails> jobDetailsFuture = restClient.sendRequest(
+			restClusterClientConfiguration.getRestServerAddress(),
+			restClusterClientConfiguration.getRestServerPort(),
+			headers
+		);
+		return jobDetailsFuture
+			.thenApply(MultipleJobsDetails::getRunning);
```

Comment:
I think we also have to return the list of completed jobs not only the running.

Reply:
we don't _have_ to, as the LIST command only cares about running jobs, but we certainly could just return all. The CLI is filtering the jobs anyway.